tests/test_rules.py

```diff
@@ -0,0 +1,273 @@
+    msg = "The 'get_overrides' method is deprecated. Use 'get_rules' instead."
+    with pytest.warns(DeprecationWarning, match=msg):
+        rules = default_registry.get_overrides()
+
+    # It should still work as usual
+    assert len(rules) == 15
+
+
+def test_consume_module_not_existing() -> None:
+    with pytest.raises(ImportError):
+        consume_modules("this_does_not_exist")
+
+
+def test_list_page_objects_all_consume() -> None:
+    """A test similar to the one above but calls ``consume_modules()`` to properly
+    load the ``@handle_urls`` decorators from other modules/packages.
+    """
+    consume_modules("tests_extra")
+    rules = default_registry.get_rules()
+    page_objects = {po.use for po in rules}
+    assert any(["po_lib_sub_not_imported" in po.__module__ for po in page_objects])
+
+
+def test_registry_search_rules() -> None:
+    rules = default_registry.search_rules(use=POTopLevel2)
+    assert len(rules) == 1
+    assert rules[0].use == POTopLevel2
+
+    rules = default_registry.search_rules(instead_of=POTopLevelOverriden2)
+    assert len(rules) == 1
+    assert rules[0].instead_of == POTopLevelOverriden2
+
+    # Such rules doesn't exist
+    rules = default_registry.search_rules(use=POModuleOverriden)
+    assert len(rules) == 0
+
+
+def test_registry_search_overrides_deprecation() -> None:
+    msg = "The 'search_overrides' method is deprecated. Use 'search_rules' instead."
+    with pytest.warns(DeprecationWarning, match=msg):
+        rules = default_registry.search_overrides(use=POTopLevel2)
+
+    # It should still work as usual
+    assert len(rules) == 1
+    assert rules[0].use == POTopLevel2
+
+
+def test_from_apply_rules() -> None:
+    rules = [
+        ApplyRule(
+            for_patterns=Patterns(include=["sample.com"]),
+            use=POTopLevel1,
+            instead_of=POTopLevelOverriden2,
+        )
+    ]
+
+    registry = PageObjectRegistry.from_apply_rules(rules)
+
+    assert registry.get_rules() == rules
+    assert default_registry.get_rules() != rules
+
+
+def test_from_override_rules_deprecation() -> None:
+    rules = [
+        ApplyRule(
+            for_patterns=Patterns(include=["sample.com"]),
+            use=POTopLevel1,
+            instead_of=POTopLevelOverriden2,
+        )
+    ]
+
+    msg = (
+        "The 'from_override_rules' method is deprecated. "
+        "Use 'from_apply_rules' instead."
+    )
+    with pytest.warns(DeprecationWarning, match=msg):
+        registry = PageObjectRegistry.from_override_rules(rules)
+
+    assert registry.get_rules() == rules
+    assert default_registry.get_rules() != rules
+
+
+def test_handle_urls_deprecation() -> None:
+
+    with warnings.catch_warnings(record=True) as w:
+
+        @handle_urls("example.com", overrides=CustomProductPage)
+        class PageWithDeprecatedOverrides:
+            ...
+
+    w = [x for x in w if x.category is DeprecationWarning]
+    assert len(w) == 1
+    assert str(w[0].message) == (
```

Comment:
Can `with pytest.warns` be used to simplify it?

Reply:
Good catch! I'm not sure was this test didn't even use that 😅 Updated in https://github.com/scrapinghub/web-poet/pull/84/commits/88c511d51cfdf64f1b4ae69563f7a97f8cf61cff.